info-compiler/src/main/java/com/google/sps/webcrawler/RelevancyChecker.java

```diff
@@ -41,10 +41,13 @@
    *   lack of permission to access the library.
    */
   public RelevancyChecker() throws IOException {
-    this.languageServiceClient = LanguageServiceClient.create();
+    this(LanguageServiceClient.create());
   }
 
-  public RelevancyChecker(LanguageServiceClient languageServiceClient) {
+  /**
+   * For testing purposes.
+   */
+  RelevancyChecker(LanguageServiceClient languageServiceClient) {
```

Comment:
You shouldn't be creating methods just for testing purposes .Instead,  redesign this class so that the constructor takes a languageServiceClient object. If the object is null, you can call LanguageServiceClient.create(), otherwise use the one that's passed in

Reply:
Got it!
To confirm: For Java, other instantiations of RelevancyChecker must pass in null explicitly: `new RelevancyChecker(null)` Would this be desirable?

Reply:
@suparnabarua  I think it's fine to have duplicate constructors, with one for testing.  I've definitely done this in my work at Google.

Reply:
as per offline discussion, keep it as it is

Reply:
got it! thank you so much!